idl/testing/run_tests.py

```diff
@@ -1213,6 +1325,119 @@ def idl_test(code, expected_output):
 
     ''')
 
+
+#---------------------------------------------------------------------------
+# Read: various permutations of reading data from a tree
+idl_test(f'''
+            
+testid = 'IDL-read-various'         
+
+mdsconnect, '{server}'
+mdsopen, '{tree}', {shot}
+print, mdsvalue('{ascii}')
+print, mdsvalue('{numeric}')
+print, mdsvalue('{signal}')
+print, mdsvalue('{fullpath}')
+mdsset_def, '{relative_def}'
+print, mdsvalue('{relative}')
+mdsset_def, '{reset_def}'
+print, mdsvalue('{tag}')
+print, mdsvalue('{wildcard}')
+print, mdsvalue('{getnci}')
+print, mdsvalue('{rank}')
+print, mdsvalue('{ndims}')
+print, mdsvalue('{dim_of}')
+print, mdsvalue('{units_of}')
+mdsclose, '{tree}', '{shot}'
+
+''',
+f'''
+
+% Compiled module: MDSCONNECT.
+% Compiled module: MDS_KEYWORD_SET.
+% Compiled module: MDSDISCONNECT.
+% Compiled module: MDSOPEN.
+% Compiled module: MDSVALUE.
+% Compiled module: MDSCHECKARG.
+% Compiled module: MDSISCLIENT.
+{ascii_value}
+{numeric_value}
+{signal_value}
+{fullpath_value}
+% Compiled module: MDSSET_DEF.
+% Compiled module: MDSSETDEFAULT.
+{relative_value}
+{tag_value}
+{wildcard_value}
+{getnci_value}
+{rank_value}
+{ndims_value}
+{dim_of_value}
+{units_of_value}
+% Compiled module: MDSCLOSE.
+
+''')
+
+
+#---------------------------------------------------------------------------
+# Write: various permutations of writing data to a tree
+#
+# Note: Uses local tree on the build server (i.e., not using mdsip).
+#       The $default_tree_path must point to the "idl/testing" directory.
+idl_test(f'''
+            
+testid = 'IDL-write-various'         
+
+mdsopen, '{write_tree}', '{write_shot}'
+mdsput, 'A_TEXT', ' "string_a" '
+mdsput, 'B_NUM', '22'
+mdsput, 'C_SIGNAL', 'build_signal([10,-10,5,-5,0],[10.2,-10.2,5.4,-5.4,0.0], [0 .. 4])'
+mdsput, '\TOP.SUBTREE_1:D_TEXT', ' "string_d" '
+mdsput, 'SUBTREE_1.E_UNITS', 'build_with_units(55, "volts")'
+mdsset_def, 'SUBTREE_1'
+mdsput, '.-.SUBTREE_2:F_NUM', '66'
+mdsput, '\TAG_G', '77'
```

Comment:
Do you ever try reading the data from this tag (or just some tag)? I looked and I don't think I saw something but I could have missed it.

Reply:
I believe in line 1412 you read the data stored here, but it's not referenced by the tag specifically.

Reply:
Hi @ModestMC -- In the `IDL-read-various` function at line 1344 is a read using a tag.

You are correct that in the `IDL-write-various` function the value written by `\TAG_G` is checked at line 1412 to confirm that the write by tag succeeded.

Reply:
The line numbers changed significantly when switched to the `argparse` template.   The `IDL-read-various` read of a tag is now around line 1518.   And the `IDL-write-various` check of a tag is around line 1586.